src/charm.py

```diff
@@ -201,7 +216,7 @@ def _pebble_layer(self, environment: Dict) -> Layer:
                     SERVICE_NAME: {
                         "override": "replace",
                         "summary": "opentelemetry-collector-k8s service",
-                        "command": f"/usr/bin/otelcol --config={CONFIG_PATH}",
+                        "command": " ".join(filter(None, ("/usr/bin/otelcol", f"--config={CONFIG_PATH}", feature_gates_flag))),
```

Comment:
Maybe extract the command to somewhere else, since it's also repeated in `_pebble_checks()`? This way maybe you won't have to pass `feature_gates` to it